dali/image/tiff.h

```diff
@@ -17,10 +17,13 @@
 
 #include <vector>
 #include <string>
+#include <utility>
+#include <memory>
 #include "dali/image/generic_image.h"
 
 namespace dali {
 
+namespace legacy_impl {
```

Comment:
We can get rid of this. It's not really legacy implementation, it's just different implementation, which will be working whenever `DALI_USE_LIBTIFF` is false, which is perfectly legal build

Reply:
ok